src/csharp/Microsoft.Spark/Sql/Types/Timestamp.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Spark.Sql.Types
+{
+    /// <summary>
+    /// Represents Timestamp containing year, month, day, hour, minute, second, microsecond in
+    /// Coordinated Universal Time (UTC).
+    /// </summary>
+    public class Timestamp
+    {
+        private readonly DateTime _dateTime;
+        private static readonly DateTime s_unixTimeEpoch =
+            new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        /// <summary>
+        /// Constructor for Timestamp class.
+        /// </summary>
+        /// <param name="dateTime">DateTime object</param>
+        public Timestamp(DateTime dateTime)
+        {
+            _dateTime = dateTime.ToUniversalTime();
+
+            if (dateTime.Kind != DateTimeKind.Utc)
```

Comment:
You can check this before the above line. Can you also add a test for this?

Reply:
Added this in Unit test.